pkg/volume/util/subpath/subpath_linux.go

```diff
@@ -260,7 +260,7 @@ func doCleanSubPaths(mounter mount.Interface, podDir string, volumeName string)
 
 		// scan /var/lib/kubelet/pods/<uid>/volume-subpaths/<volume>/<container name>/*
 		fullContainerDirPath := filepath.Join(subPathDir, containerDir.Name())
-		err = filepath.Walk(fullContainerDirPath, func(path string, info os.FileInfo, _ error) error {
+		err = filepath.WalkDir(fullContainerDirPath, func(path string, info os.DirEntry, _ error) error {
```

Comment:
@msau42 do we only care about the files inside `/var/lib/kubelet/pods/<uid>/volume-subpaths/<volume>/<container name>/*` if so wouldn't `Readdir` work? I'm not sure why we need to to a traversal

Reply:
Good question. I traced back to when we changed it from `Readdir` to `Walk` and found myself :-)

https://github.com/kubernetes/kubernetes/pull/71804

The issue this was fixing was a subpath on an nfs server that became stale. What happened was that `Readdr` on a stale nfs mount would return an error, whereas `Walk` would pass through the error.

It may be worth adding a comment here explaining that for future readers. Also good to check if the e2e test added for the stale nfs issue is passing with this fix.

Reply:
e2e test "Kubernetes e2e suite: [sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [LinuxOnly]" passed
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/107669/pull-kubernetes-e2e-gce-ubuntu-containerd/1484417919810015232/

Reply:
Thanks for all your patience! I have added a detail comment trying to make it more clear.